src/Infrastructure/VM/src/ESMCI_VMKernel.C

```diff
@@ -2597,7 +2597,12 @@ void *VMK::startup(class VMKPlan *vmp, void *(fctp)(void *, void *),
     if (pos != string::npos){
       // found wildcard -> replace with local pet number
       int digits = (int) log10(new_npets);    // always safe
-      if (digits * 10 != new_npets) ++digits; // correct number of digits
+      // correct the number of digits
+      int test_npets = 1;
+      for (int j=0; j<digits; j++)
+        test_npets *= 10;
+      if (test_npets != new_npets) ++digits;
```

Comment:
Maybe you might want to confider to combine these two if blocks since they are doing same thing. Just an idea.